packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/Messages.java

```diff
@@ -264,6 +360,324 @@ ArrayList<Object> toList() {
     }
   }
 
+  /**
+   * Pigeon version of Java ProductDetails.OneTimePurchaseOfferDetails.
+   *
+   * <p>Generated class from Pigeon that represents data sent in messages.
+   */
+  public static final class PlatformOneTimePurchaseOfferDetails {
+    private @NonNull Long priceAmountMicros;
+
+    public @NonNull Long getPriceAmountMicros() {
+      return priceAmountMicros;
+    }
+
+    public void setPriceAmountMicros(@NonNull Long setterArg) {
+      if (setterArg == null) {
+        throw new IllegalStateException("Nonnull field \"priceAmountMicros\" is null.");
```

Comment:
setPrinceAmountMicros I think is only called by in plugin java code so this would be a plugin author error. Is that why this is not a PlatfromException?

Reply:
Pigeon generates these errors for the native classes in Java so that if someone violates their own API constraints in native plugin code it'll create an error on the Java side, where it happened, rather than the Dart side (when it's read) further from the mistake.